### PR TITLE
Improve the MatrixFree doxygen formatting

### PR DESCRIPTION
**Pull request by drwells**
This is a followup to #7129: I added name qualifications and a few other minor things to make it easier to read the matrix free module header.

### CODE REVIEW

File: include/deal.II/grid/tria_iterator.h [hunk 1]
```diff
@@ -165,10 +165,10 @@ class TriaActiveIterator;
  *
  * The accessors provided by the library consists of two groups, determined by
  * whether they access the data of Triangulation objects or
- * DoFHandler/hp::DoFHandler objects. They are derived from TriaAccessor and
- * DoFAccessor, respectively. Each group also has specialized accessors for
- * cells (as opposed to faces and lines) that offer more functionality such as
- * accessing neighbors.
+ * DoFHandler&thinsp;-&thinsp;hp::DoFHandler objects. They are derived from
```
Comment (masterleinad) [inline include/deal.II/grid/tria_iterator.h]: That looks uncommon. Why do you want to use `-` instead of `/`?

Comment (drwells) [reply include/deal.II/grid/tria_iterator.h]: I thought the opposite: hyphens are for linking words together in prose. I don't think I have seen `/`s used to do that.

**Comment by drwells** [comment]
I fixed the `internal::MatrixFreeFunctions::tensor_symmetric_hermite` link.

File: include/deal.II/grid/tria_iterator.h [hunk 2]
```diff
@@ -165,10 +165,10 @@ class TriaActiveIterator;
  *
  * The accessors provided by the library consists of two groups, determined by
```
Comment (bangerth) [inline include/deal.II/grid/tria_iterator.h]: while you're here, can you replace "consists" by "consist" to get the grammar right? Though maybe "fall" or "can be grouped into two classes" would actually be better

File: include/deal.II/grid/tria_iterator.h (hunk 1 above)
Comment (drwells) [reply include/deal.II/grid/tria_iterator.h]: I should get a style manual and read it. Wikipedia says that `/` can be used as a shorthand for 'and' so lets stick with `/` here; you're right.

**Pull request merged**